--- gcc/libiberty/simple-object.c	2018-04-13 19:52:10.000000000 +0300
+++ gcc/libiberty/simple-object.c.patched	2019-02-04 12:01:37.000000000 +0200
@@ -44,6 +44,10 @@
 #define SEEK_SET 0
 #endif
 
+#ifndef O_BINARY
+#define O_BINARY 0
+#endif
+
 #include "simple-object-common.h"
 
 /* The known object file formats.  */
@@ -326,7 +330,7 @@
       return errmsg;
     }
 
-  outfd = creat (dest, 00777);
+  outfd = open (dest, O_CREAT|O_WRONLY|O_TRUNC|O_BINARY, 00777);
   if (outfd == -1)
     {
       *err = errno;
--- gcc/gcc/gcc.c	2018-02-09 13:52:37.000000000 +0200
+++ gcc/gcc/gcc.c.patched	2019-02-13 16:08:05.000000000 +0200
@@ -7738,7 +7738,11 @@
 				    X_OK, false);
   if (lto_wrapper_file)
     {
+#if defined (__MINGW32__)
+      // Win32 fails to CreateProcess if spaces are escaped.
+#else
       lto_wrapper_file = convert_white_space (lto_wrapper_file);
+#endif
       lto_wrapper_spec = lto_wrapper_file;
       obstack_init (&collect_obstack);
       obstack_grow (&collect_obstack, "COLLECT_LTO_WRAPPER=",
@@ -8330,7 +8334,11 @@
 		fatal_error (input_location,
 			     "-fuse-linker-plugin, but %s not found",
 			     LTOPLUGINSONAME);
+#if defined (__MINGW32__)
+	      linker_plugin_file_spec = temp_spec;
+#else
 	      linker_plugin_file_spec = convert_white_space (temp_spec);
+#endif
 	    }
 #endif
 	  lto_gcc_spec = argv0;
